src/pytorch_lightning/utilities/argparse.py

```diff
@@ -60,7 +62,7 @@ def from_argparse_args(
         >>> trainer = Trainer.from_argparse_args(args, logger=False)
     """
     if isinstance(args, ArgumentParser):
-        args = cls.parse_argparser(args)
+        args = cls.parse_argparser(args)  # type: ignore[union-attr]
```

Comment:
I have to say I'm not sure about these `# type: ignore`s. I'm gonna have to think about it a bit

Reply:
resolved in https://github.com/Lightning-AI/lightning/pull/13693/commits/517576db4d0b1cbda58bc1ac63416436e8ddd198